src/sentry/tasks/process_buffer.py

```diff
@@ -54,15 +56,33 @@ def process_pending_batch() -> None:
 
 
 @instrumented_task(name="sentry.tasks.process_buffer.process_incr", queue="counters-0")
-def process_incr(**kwargs):
+def process_incr(
+    model: type[models.Model] | None,
+    columns: dict[str, int],
+    filters: dict[str, Any],
+    extra: dict[str, Any] | None = None,
+    signal_only: bool | None = None,
+    model_name: str | None = None,
+    **kwargs,
+):
     """
     Processes a buffer event.
     """
     from sentry import buffer
 
-    sentry_sdk.set_tag("model", kwargs.get("model", "Unknown"))
+    if not model:
+        assert model_name, "model or model_name is required"
+        model = apps.get_model(model_name)
```

Comment:
Don't we need an `app_name` here too?

Reply:
The docs indicate that you can send `app_label.model_name` to `get_model()` and get the right result.

Reply:
Could we add an assert to validate that we're passing a string with a `.`? And/or maybe a comment saying that `model_name` should include the app label?

Reply:
Good idea. Added an assert